activesupport/lib/active_support/actionable_error.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module ActiveSupport
+  module ActionableError
+    extend Concern
+
+    NonActionable = Class.new(StandardError)
+
+    included do
+      class_attribute :_actions, default: Hash.new { |_, label|
+        raise NonActionable, "Cannot find action \"#{label}\" for #{self}"
+      }
+    end
+
+    def self.===(other)
+      super || Module === other && other.ancestors.include?(self)
```

Comment:
🤔 I don't understand the necessity of the `Module === other` part.

Reply:
This is a subtle trick:

I'm using the hierarchy to check whether an error is actionable with the equivalent of an `#is_a?` check. For example: `[].is_a?(Enumerable)` (`Enumerable === []`), because `Enumerable` is in `Array.ancestors` chain. This check, however, expects an instance. I sometimes the `#===` to check for actionability of error classes and not error instances when dispatching methods. For this case, I extended the case operator to check for actionable classes and the `Module === other` makes sure I can call `#ancestors` later on.